### ISSUE TITLE
Course Schedules DST Issue

### ISSUE THREAD

**Issue by alexdlaird** [HeliumEdu/platform#368]
When a course schedule spans the DST boundary, if a User looks ahead in their schedule (meaning the current date is prior to DST, but the week they are looking at on the calendar is after), the class time will reflect one after off.

This can be proven with a unit test as well, in `testcasecoursescheduleresourceviews.py`, the `test_get_course_schedule_as_events` test already has a class schedule that crosses the DST boundary. Adding these assertions to the end of that test then should pass, but it fails:

```py
        self.assertEqual(response.data[-1]['start'], '2017-05-08T10:30:00Z')
        self.assertEqual(response.data[-1]['end'], '2017-05-08T13:00:00Z')
```

This could be due to use `make_aware` in `coursescheduleservice.py::_create_events_from_course_schedules()`, but unfortunately simply removing this will cause other side effects, so we need to think of a better solution.

**Comment by alexdlaird** [comment]
The same issue exists for iCal feeds being streamed from an external calendar (ie. Google Calendar) to display in Helium. These are the only two points `make_aware` is used in the codebase, so certainly that is the issue, we need to be applying the TZ before calling `make_aware` (or perhaps not using `make_aware at all and just applying the TZ).

**Comment by alexdlaird** [comment]
This has been resolved with release `1.4.50`.
